src/VehicleSelect/VehicleSelect.tsx

```diff
@@ -0,0 +1,221 @@
+import Autocomplete from "../Autocomplete";
+import { Box } from "@mui/material";
+import React from "react";
+import { VehicleSelectProps } from "./VehicleSelect.types";
+
+// component to select a vehicle
+function VehicleSelect({
+  allVehicles = [],
+  selectedVehicles = [],
+  flexDirection = "column",
+  flexWrap = "nowrap",
+  allGates = [],
+  onVehicleChange = () => {}
+}: VehicleSelectProps) {
+  // derive state for project
+  const selectedProjects = [
+    ...new Set(selectedVehicles.map(vehicle => vehicle.project))
+  ];
+  if (selectedProjects.length > 1)
+    throw new Error("Project selection is ambiguous");
+  const selectedProject = selectedProjects[0] ?? "";
+
+  // derive all projects
+  const allProjects = [
+    ...new Set(allVehicles.map(vehicle => vehicle.projectCode))
+  ].sort();
+
+  // derive state for model year
+  const selectedModelYears = [
+    ...new Set(selectedVehicles.map(vehicle => vehicle.modelYear))
+  ];
+  if (selectedModelYears.length > 1)
+    throw new Error("Multiple year is ambiguous");
+  const selectedModelYear = selectedModelYears[0] ?? "";
+
+  // derive all model years
+  const allModelYears = [
+    ...new Set(
+      allVehicles
+        .filter(v => v.projectCode === selectedProject)
+        .map(vehicle => vehicle.modelYear)
+    )
+  ].sort();
+
+  // derive state for variant
+  const selectedVariants = [
+    ...new Set(selectedVehicles.map(vehicle => vehicle.variant))
+  ]
+    .filter(v => v !== "")
+    .sort();
+
+  // derive all variants
+  const allVariants = [
+    ...new Set(
+      allVehicles
+        .filter(
+          v =>
+            v.projectCode === selectedProject &&
+            v.modelYear === selectedModelYear
+        )
+        .map(v => v.variant)
+    )
+  ].sort();
+
+  // derive state for gate
+  const selectedGates = [
+    ...new Set(selectedVehicles.map(vehicle => vehicle.gate))
+  ]
+    .filter(v => v !== "")
+    .sort();
+
+  return (
+    <Box
+      sx={{
+        display: "flex",
+        flexDirection,
+        flexWrap,
+        gap: flexDirection === "row" ? "0 24px" : 0
+      }}
+    >
+      <Box flex="40%">
```

Comment:
Instead of having fixed flex=40% can we have that as a prop? so we can control it from apps.

Reply:
There is already flexibility with this. We can have 2 or 4 fields in a row.